mr/src/test/java/org/apache/iceberg/mr/mapreduce/TestIcebergInputFormat.java

```diff
@@ -423,70 +377,147 @@ public Catalog apply(Configuration conf) {
+    }
+
+    private TestInputFormat(List<IcebergSplit> splits, List<T> records) {
+      this.splits = splits;
+      this.records = records;
+    }
+
+    public List<T> getRecords() {
+      return records;
+    }
+
+    public List<IcebergSplit> getSplits() {
+      return splits;
+    }
+
+    public void validate(List<T> expected) {
+      Assert.assertEquals(expected, records);
+    }
   }
 
-  private static <T> List<T> readRecords(Configuration conf) {
-    TaskAttemptContext context = new TaskAttemptContextImpl(conf, new TaskAttemptID());
-    IcebergInputFormat<T> icebergInputFormat = new IcebergInputFormat<>();
-    List<InputSplit> splits = icebergInputFormat.getSplits(context);
-    return
-        FluentIterable
-            .from(splits)
-            .transformAndConcat(split -> readRecords(icebergInputFormat, split, context))
-            .toList();
+  private static class TestMapRedInputFormat<T> extends TestInputFormat<T> {
+
+    private TestMapRedInputFormat(List<IcebergSplit> splits, List<T> records) {
+      super(splits, records);
+    }
+
+    private static<T> TestMapRedInputFormat<T> create(Configuration conf) throws IOException {
+      JobConf job = new JobConf(conf);
+      MapredIcebergInputFormat<T> inputFormat = new MapredIcebergInputFormat<>();
+      org.apache.hadoop.mapred.InputSplit[] splits = inputFormat.getSplits(job, 1);
+
+      List<IcebergSplit> iceSplits = new ArrayList<>(splits.length);
+      List<T> records = new ArrayList<>();
+
+      for (org.apache.hadoop.mapred.InputSplit split : splits) {
+        iceSplits.add((IcebergSplit) split);
+        org.apache.hadoop.mapred.RecordReader<Void, Container<T>> reader = inputFormat.getRecordReader(split, job, Reporter.NULL);
+
+        try {
+          Container<T> container = reader.createValue();
+
+          while (reader.next(null, container)) {
+            records.add(container.get());
+          }
+        } finally {
+          reader.close();
+        }
+      }
+
+      return new TestMapRedInputFormat<>(iceSplits, records);
+    }
+
   }
 
-  private static <T> Iterable<T> readRecords(
-      IcebergInputFormat<T> inputFormat, InputSplit split, TaskAttemptContext context) {
-    RecordReader<Void, T> recordReader = inputFormat.createRecordReader(split, context);
-    List<T> records = new ArrayList<>();
-    try {
-      recordReader.initialize(split, context);
-      while (recordReader.nextKeyValue()) {
-        records.add(recordReader.getCurrentValue());
+  private static class TestMapReduceInputFormat<T> extends TestInputFormat<T> {
+
+    private TestMapReduceInputFormat(List<IcebergSplit> splits, List<T> records) {
+      super(splits, records);
+    }
+
+    private static<T> TestMapReduceInputFormat<T> create(Configuration conf) throws IOException {
+      TaskAttemptContext context = new TaskAttemptContextImpl(conf, new TaskAttemptID());
+      IcebergInputFormat<T> inputFormat = new IcebergInputFormat<>();
```

Comment:
As mentioned before, I think to be consistent with the naming we could have  `TestMapReduceIcebergInputFormat` for `MapReduceIcebergInputFormat`.